File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    rna_trans = {'G': 'C',
                 'C': 'G',
                 'T': 'A',
                 'A': 'U'}

    rna = []
    for nucleotide in dna_strand:
        if nucleotide not in rna_trans:
            raise ValueError("DNA strand contains invalid nucleotide(s)")
        else:
            rna += rna_trans[nucleotide]

    return ''.join(rna)

File: go/rna-transcription/rna_transcription.go
package strand

var rnaMap = map[byte]byte{
	'G': 'C',
	'C': 'G',
	'T': 'A',
	'A': 'U',
}

func ToRNA(dna string) string {
	rna := make([]byte, len(dna))
	for i, v := range []byte(dna) {
		rna[i] = rnaMap[v]
	}
	return string(rna)
}
